packages/react-client/src/ReactFlightClient.js

```diff
@@ -138,10 +151,23 @@ export function reportGlobalError(response: Response, error: Error): void {
   });
 }
 
-function definePendingProperty(
+function readMaybeChunk<T>(maybeChunk: Chunk<T> | T): T {
+  if ((maybeChunk: any).$$typeof !== CHUNK_TYPE) {
```

Comment:
I've noticed `maybeChunk` is literally the `User` function in the Relay test. Is this intentional? We're missing some mocking that represents the module transport there?

Reply:
It wasn't intentional at first but I did notice it while writing the tests and decided to keep it. I think we should do some mocking here but we don't have it wired up yet so we're not sure what we're simulating yet.

Also, the tests are far from sufficient yet. Needs more async stuff. I need to write more but don't want to block on it.